StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/SavedPaymentMethodCollectionView.swift

```diff
@@ -372,6 +407,42 @@ extension SavedPaymentMethodCollectionView {
                 }()
             }
         }
+
+        private func addDefaultBadge() {
+            defaultBadge.isHidden = true
+            if defaultBadge.superview == nil {
+                defaultBadge.translatesAutoresizingMaskIntoConstraints = false
+                contentView.addSubview(defaultBadge)
+                labelHeightConstraint = label.heightAnchor.constraint(equalToConstant: 20)
+                guard let labelBottomConstraint, let labelHeightConstraint else { return }
+                NSLayoutConstraint.deactivate([
+                    labelBottomConstraint
+                ])
+                NSLayoutConstraint.activate([
+                    labelHeightConstraint,
+                    defaultBadge.topAnchor.constraint(
+                        equalTo: label.bottomAnchor, constant: 4),
+                    defaultBadge.bottomAnchor.constraint(equalTo: contentView.bottomAnchor),
+                    defaultBadge.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 2),
+                    defaultBadge.trailingAnchor.constraint(equalTo: contentView.trailingAnchor)
+                ])
+                contentView.layoutIfNeeded()
+            }
+        }
+
+        private func removeDefaultBadge() {
+            if defaultBadge.superview != nil {
+                guard let labelHeightConstraint, let labelBottomConstraint else { return }
+                NSLayoutConstraint.deactivate([
+                    labelHeightConstraint
+                ])
+                defaultBadge.removeFromSuperview()
+                NSLayoutConstraint.activate([
+                    labelBottomConstraint
+                ])
+                contentView.layoutIfNeeded()
+            }
+        }
```

Comment:
I think a simpler approach would be to add the default badge to all PMs, then just hide/show it when needed.
This way you can use one size for all the cells and not need to swap the cell sizes, and can move adding the default badge up to where the rest of the cell adds it's view's, and you won't need to track the extra constraints. You may also not need to reload the collection view either.

Reply:
If you look in the Figma, the sheet gets taller when entering edit mode due to the default badge. The way it currently works, it updates the item size and intrinsic content size and adds the default badge to all PMs upon entering edit mode and hides/shows when needed. When it exits edit mode, the default badge is removed from all the cells and the item size and intrinsic content size return to normal. If the default badge is there the whole time and the item size and intrinsic content sizes are constant the whole time, then there is extra space under the PMs, and the height of the sheet doesn't change.

Reply:
I've figured out that I can have the default badge added to the view but change the constraints so that it won't take up the extra space, but I don't see a way around the dynamically changing cell sizes yet